domain/unitstate/service/service.go

```diff
@@ -102,3 +121,13 @@ func (s *Service) SetState(ctx context.Context, as unitstate.AgentState) error {
 		return nil
 	})
 }
+
+// GetState returns the full agent state of the unit.
+// If no unit with the uuid exists, a [errors.UnitNotFound] error is returned.
+func (s *Service) GetState(ctx context.Context, uuid string) (unitstate.AgentState, error) {
+	state, err := s.st.GetUnitState(ctx, uuid)
+	if err != nil && !errors.Is(err, unitstateerrors.UnitStateEmpty) {
```

Comment:
todo: add a comment about why it's okay to return no error on unitstateerrors.UnitStateEmpty.

Reply:
I decided actually to just remove this error and comment on the functions instead.